src/Sylius/Bundle/ApiBundle/Controller/TerminateCatalogPromotionAction.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\TerminateCatalogPromotion;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class TerminateCatalogPromotionAction
+{
+    public function __construct(
+        private MessageBusInterface $commandBus,
+        private RepositoryInterface $catalogPromotionRepository,
+    ){
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $code = $request->attributes->get('code');
+
+        /** @var CatalogPromotionInterface|null $catalogPromotion */
+        $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $code]);
+
+        if (null === $catalogPromotion) {
+            return new Response(status: Response::HTTP_NOT_FOUND);
+        }
+
+        $this->commandBus->dispatch(new TerminateCatalogPromotion($code));
+
+        return new Response(status: Response::HTTP_NO_CONTENT);
```

Comment:
I left a 204 status code due to the behat response checker which's looking for 204 indeed. I wonder if we should return 204 or 202 depending on messenger sync / async config, wdyt?